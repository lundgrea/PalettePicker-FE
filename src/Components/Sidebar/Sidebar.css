.Sidebar {
  display: flex;
  flex-direction: column;
  width: 25%;
  height: 100vh;
  border: 2px solid black;
  /* font-family: 'Caveat', cursive; */
}