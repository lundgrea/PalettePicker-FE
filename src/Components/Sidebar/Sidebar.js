import React from "react";
import FolderForm from "../../Components/FolderForm/FolderForm";
import FoldersContainer from "../../Components/FoldersContainer/FoldersContainer";
import PropTypes from 'prop-types';
import "./Sidebar.css";

export const Sidebar = props => {
  return (
    <aside className="Sidebar">
      <FolderForm 
      folders={props.folders} 
      addNewFolder={props.addNewFolder} 
      getAllFolders={props.getAllFolders}/>
      <FoldersContainer 
      folders={props.folders} 
      deleteAFolder={props.deleteAFolder}
      getAFoldersPalettes={props.getAFoldersPalettes}/>
    </aside>
  );
};

export default Sidebar;


Sidebar.propTypes = {
  addNewFolder:PropTypes.func,
  deleteAFolder: PropTypes.func,
  folder: PropTypes.array,
  getAFoldersPalette: PropTypes.func,
  getAllFolders: PropTypes.func
};