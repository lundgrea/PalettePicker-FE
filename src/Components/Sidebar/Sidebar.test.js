import React from "react";
import { shallow } from "enzyme";
import { Sidebar } from "./Sidebar";

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

let wrapper;

const folders = []

beforeEach(() => {
  wrapper = shallow(<Sidebar folders={folders} addNewFolder={jest.fn()} getAllFolders={jest.fn()} getAFoldersPalettes={jest.fn()} deleteAFolder={jest.fn()}/>);
});

describe("Sidebar", () => {
  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

