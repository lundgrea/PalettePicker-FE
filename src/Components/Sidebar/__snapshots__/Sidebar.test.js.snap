// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should match snapshot 1`] = `
<aside
  className="Sidebar"
>
  <FolderForm
    addNewFolder={[MockFunction]}
    folders={Array []}
    getAllFolders={[MockFunction]}
  />
  <FoldersContainer
    deleteAFolder={[MockFunction]}
    folders={Array []}
    getAFoldersPalettes={[MockFunction]}
  />
</aside>
`;
